### ISSUE TITLE
Error in formatC(30.5542270992922, format = "f", digits = 2L, as.output = FALSE) :

### ISSUE THREAD

**Issue by pfrombris** [NumbersInternational/flipMultivariates#16]
After having huge issues downloading and installing this package, as several previous commentators have, I finally managed.

After figuring out that the response variable for LDA has to be a factor, every time I use LDA I get:

Error in formatC(30.5542270992922, format = "f", digits = 2L, as.output = FALSE) :

The number changes, depending on which combination of factors I include.

I have tried using the vehicles dataset, as per the online example and get the same error message.

Please can you tell me how to avoid this error and also tell me where there is a vignette or other documentation about this package.

**Comment by mwmclean** [comment]
Sorry for the difficulties you experienced installing the package. I'm not able to reproduce the issue you're describing using `LDA` and the vehicles data on our servers. Can you please share the code you're using and the output of `sessionInfo()`?

There is a blog post [here](https://www.displayr.com/linear-discriminant-analysis-in-r-an-introduction/), which it sounds like you already found and documentation for the function arguments and some references in the LDA help file, `?LDA`. The package does not contain any vignettes.

**Comment by pfrombris** [comment]
Hi,

Thank you so much for getting back to me so promptly.

This is the code, and below the console output from RStudio and below that some details of my system:

library(dplyr)
library(flipMultivariates)
vehicles <- read.csv("D://documents/downloads/vehicle_csv.csv")
vehicles$Class <- as.factor(vehicles$Class)
lda <- LDA(Class ~ ., data = vehicles)
lda

> library(dplyr)
> library(flipMultivariates)
> vehicles <- read.csv("D://documents/downloads/vehicle_csv.csv")
> vehicles$Class <- as.factor(vehicles$Class)
> lda <- LDA(Class ~ ., data = vehicles)
> lda
Error in formatC(41.0500162669195, format = "f", digits = 2L, as.output = FALSE) : 
  unused argument (as.output = FALSE)

-----------------------------------------------------------------------------------------------------------------
R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] dplyr_1.0.7             flipMultivariates_1.1.4

Please let me know if you need any more information.

kind regards,

Peter

**Comment by pfrombris** [comment]
Hi Matthew,

I posted this reponse, not sure f you've seen it?


Hi,


Thank you so much for getting back to me so promptly.


This is the code, and below the console output from RStudio and below that some details of my system:



library(dplyr)
library(flipMultivariates)
vehicles <- read.csv("D://documents/downloads/vehicle_csv.csv")
vehicles$Class <- as.factor(vehicles$Class)
lda <- LDA(Class ~ ., data = vehicles)
lda



>
>
> library(dplyr)
> library(flipMultivariates)
> vehicles <- read.csv("D://documents/downloads/vehicle_csv.csv")
> vehicles$Class <- as.factor(vehicles$Class)
> lda <- LDA(Class ~ ., data = vehicles)
> lda
> Error in formatC(41.0500162669195, format = "f", digits = 2L, as.output = FALSE) :
> unused argument (as.output = FALSE)
>
>
>




R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)




attached base packages:
[1] stats graphics grDevices utils datasets methods
[7] base




other attached packages:
[1] dplyr_1.0.7 flipMultivariates_1.1.4



Please let me know if you need any more information.


kind regards,


Peter


kind regards,

Peter

***@***.***



2 Dec 2021, 22:26 by ***@***.***:

**Comment by mwmclean** [comment]
I'm still unable to reproduce this on my Windows 10 machine with the current version of flipMultivariates from GitHub. I've used the exact data used in the blog post I've linked. Please share the csv you are using and provide the _complete_ `sessionInfo()`, including packages "loaded via a namespace (and not attached)".

**Comment by pfrombris** [comment]
Hi,

Thanks again.  Here is the complete sessioninfo() with namespace packages and the data.

Peter

***@***.***


--------------------------------------------------

R version 4.1.2 (2021-11-01) -- "Bird Hippie"


Copyright (C) 2021 The R Foundation for StatisticalComputing


Platform: x86_64-w64-mingw32/x64 (64-bit)


 


R is free software and comes with ABSOLUTELY NO WARRANTY.


You are welcome to redistribute it under certain conditions.


Type 'license()' or 'licence()' for distribution details.


 


R is a collaborative project with many contributors.


Type 'contributors()' for more information and


'citation()' on how to cite R or R packages in publications.


 


Type 'demo()' for some demos, 'help()' for on-line help, or


'help.start()' for an HTML browser interface to help.


Type 'q()' to quit R.


 


[Workspace loaded from D:/Documents/Onedrive/.RData]


 


Registered S3 methods overwritten by 'proxy':


  method               from   


  print.registry_fieldregistry


  print.registry_entryregistry


Registered S3 method overwritten by 'flipRegression':


  method      from


  print.Anova MASS


> library(dplyr)


 


Attaching package: ‘dplyr’


 


The following objects are masked from ‘package:stats’:


 


    filter, lag


 


The following objects are masked from ‘package:base’:


 


    intersect,setdiff, setequal, union


 


> library(flipMultivariates)


> vehicles <-read.csv("D://documents/downloads/vehicle_csv.csv")


> vehicles$Class <- as.factor(vehicles$Class)


> lda <- LDA(Class ~ ., data = vehicles)


> lda


Error in formatC(41.0500162669195, format = "f",digits = 2L, as.output = FALSE) :


  unused argument(as.output = FALSE)




> R.version


               _                          


platform       x86_64-w64-mingw32         


arch           x86_64                     


os             mingw32                    


system         x86_64,mingw32            


status                                    


major          4                          


minor          1.2                        


year           2021                        


month          11                         


day            01                         


svn rev        81115                      


language       R                          


version.string R version 4.1.2 (2021-11-01)


nickname       BirdHippie                 


> print(sessionInfo())


R version 4.1.2 (2021-11-01)


Platform: x86_64-w64-mingw32/x64 (64-bit)


Running under: Windows 10 x64 (build 19043)


 


Matrix products: default


 


locale:


[1] LC_COLLATE=English_United Kingdom.1252


[2] LC_CTYPE=English_United Kingdom.1252  


[3] LC_MONETARY=English_United Kingdom.1252


[4] LC_NUMERIC=C                          


[5] LC_TIME=English_United Kingdom.1252   


 


attached base packages:


[1] stats     graphics  grDevices utils     datasets  methods 


[7] base    


 


other attached packages:


[1] flipMultivariates_1.1.4 dplyr_1.0.7           


 


loaded via a namespace (and not attached):


  [1]parcoords_1.0.0              tidyr_1.1.4                 


  [3]enrichwith_0.3.1             ggplot2_3.3.5              


  [5] bit64_4.0.5                  knitr_1.36                 


  [7]dygraphs_1.1.1.6-1           multcomp_1.4-17            


  [9]data.table_1.14.2            rpart_4.1-15                


 [11]generics_0.1.1               leaflet_2.0.4.1            


 [13] callr_3.7.0                  TH.data_1.1-0              


 [15]usethis_2.1.3                mice_3.14.0                


 [17]proxy_0.4-26                 tensorflow_2.7.0            


 [19] bit_4.0.4                    xml2_1.3.3                 


 [21]lubridate_1.8.0              assertthat_0.2.1           


 [23] xfun_0.28                    hms_1.1.1                  


 [25]jquerylib_0.1.4              evaluate_0.14               


 [27] fansi_0.5.0                  readxl_1.3.1               


 [29] DBI_1.1.1                    tmvnsim_1.0-2              


 [31]htmlwidgets_1.5.4            flipStatistics_1.1.3       


 [33]stringdist_0.9.8             Rmpfr_0.8-7                 


 [35] CVXR_1.0-10                  purrr_0.3.4                


 [37]hash_2.2.6.1                 ellipsis_0.3.2             


 [39]crosstalk_1.2.0              backports_1.4.0            


 [41]insight_0.14.5               survey_4.1-1               


 [43] ROI_1.0-0                    sure_0.2.0                 


 [45] vctrs_0.3.8                  remotes_2.4.2              


 [47] abind_1.4-5                  cachem_1.0.6               


 [49] withr_2.4.3                  pryr_0.1.5                 


 [51]icarus_0.3.1                 checkmate_2.0.0            


 [53] xts_0.12.1                   prettyunits_1.1.1          


 [55]weights_1.0.4                mnormt_2.0.2               


 [57]cluster_2.1.2                pacman_0.5.1               


 [59]lazyeval_0.2.2               crayon_1.4.2               


 [61]RGoogleAnalytics_0.1.6       pkgconfig_2.0.3            


 [63] slam_0.1-49                  nlme_3.1-153               


 [65]pkgload_1.2.4                nnet_7.3-16                


 [67]flipTime_2.9.4               devtools_2.4.3             


 [69]rlang_0.4.12                 lifecycle_1.0.1            


 [71]flipTransformations_1.10.0   d3vennR_0.1.1              


 [73]sandwich_3.0-1               registry_0.5-1             


 [75]rhtmlPalmTrees_1.0.4         cellranger_1.1.0           


 [77]randomForest_4.6-14          rprojroot_2.0.2            


 [79]lmtest_0.9-39                rhtmlLabeledScatter_1.1.8  


 [81]flipRegression_1.3.45        Matrix_1.3-4               


 [83]carData_3.0-4                boot_1.3-28                


 [85] zoo_1.8-9                    base64enc_0.1-3            


 [87]formattable_0.2.1            whisker_0.4                


 [89]processx_3.5.2               png_0.1-7                  


 [91]viridisLite_0.4.0            flipTables_2.8.1           


 [93] ca_0.71.1                    flipChartBasics_2.3.1      


 [95]sparkline_2.0                ROI.plugin.lpsolve_1.0-1   


 [97]stringr_1.4.0                jpeg_0.1-9                 


 [99]scales_1.1.1                 memoise_2.0.1              


[101] magrittr_2.0.1               plyr_1.8.6                 


[103] verbs_0.13.1                 gdata_2.18.0               


[105] compiler_4.1.2               RColorBrewer_1.1-2         


[107] lme4_1.1-27.1                detectseparation_0.2       


[109] rgeolocate_1.4.1             flipData_1.5.5             


[111] snakecase_0.11.0             cli_3.1.0                  


[113] flipFormat_1.10.5            ps_1.6.0                   


[115] htmlTable_2.3.0              Formula_1.2-4              


[117] MASS_7.3-54                  mgcv_1.8-38                


[119] tidyselect_1.1.1             stringi_1.7.6              


[121] forcats_0.5.1                rhtmlPictographs_1.0.6     


[123] mitools_2.4                  latticeExtra_0.6-29        


[125] grid_4.1.2                   sass_0.4.0                 


[127] tools_4.1.2                  rhtmlDonut_1.0.6           


[129] parallel_4.1.2               rstudioapi_0.13            


[131] foreign_0.8-81               janitor_2.1.0              


[133] gridExtra_2.3                hot.deck_1.2               


[135] psycho_0.6.1                 digest_0.6.29              


[137] BiocManager_1.30.16          Rcpp_1.0.7                 


[139] car_3.0-12                   broom_0.7.10               


[141] performance_0.8.0            httr_1.4.2                 


[143] psych_2.1.9                  effects_4.2-0              


[145] colorspace_2.0-2             fs_1.5.1                   


[147] flipU_1.5.0                  reticulate_1.22            


[149] splines_4.1.2                rhtmlMetro_1.0.4           


[151] sp_1.4-6                     xgboost_1.5.0.2            


[153] plotly_4.10.0                sessioninfo_1.2.1          


[155] gmp_0.6-2.1                  jsonlite_1.7.2             


[157] nloptr_1.2.2.3               lpSolveAPI_5.5.2.0-17.7    


[159] zeallot_0.1.0                keras_2.7.0                


[161] testthat_3.1.1               R6_2.5.1                   


[163] Hmisc_4.6-0                  pillar_1.6.4               


[165] htmltools_0.5.2              mime_0.12                  


[167] glue_1.5.1                   fastmap_1.1.0              


[169] minqa_1.2.4                  DT_0.20                    


[171] flipImputation_1.0.0         class_7.3-19               


[173] codetools_0.2-18             pkgbuild_1.2.1             


[175] rhtmlMoonPlot_1.0.1          mvtnorm_1.1-3              


[177] utf8_1.2.2                   lattice_0.20-45            


[179] bslib_0.3.1                  tibble_3.1.6               


[181] flipAnalysisOfVariance_1.3.6flipStandardCharts_1.27.5  


[183] numDeriv_2016.8-1.1          curl_4.3.2                 


[185] tfruns_1.5.0                 colorRamps_2.3             


[187] gtools_3.9.2                 survival_3.2-13            


[189] striprtf_0.5.3               rmarkdown_2.11             


[191] desc_1.4.0                   munsell_0.5.0              


[193] e1071_1.7-9                  flipAPI_1.2.4              


[195] haven_2.4.3                  reshape2_1.4.4             


[197] gtable_0.3.0                 streamgraph_0.9.0          

--------------------------------------------------


6 Dec 2021, 23:36 by ***@***.***:

**Comment by mwmclean** [comment]
Can you please share the file `D://documents/downloads/vehicle_csv.csv` or provide a minimal data set using R code that reproduces the issue?

**Comment by pfrombris** [comment]
[vehicle_csv.csv](https://github.com/Displayr/flipMultivariates/files/7674645/vehicle_csv.csv)
)

**Comment by mwmclean** [comment]
Thanks. The issue you are encountering is from having a different version of `formattable` installed than the one [needed](https://github.com/Displayr/flipFormat/blob/master/DESCRIPTION#L38) by the dependency `flipFormat`. It unfortunately requires a special fork of formattable instead of the version from CRAN. If you run `remotes::install_github("Displayr/formattable")` and rerun your example it should work.

**Comment by pfrombris** [comment]
Mathew,

Thank you very much, that worked.  I had to add force=T to make it install, and restart r. I'm getting results now with my own data.  Thanks again.

kr

Peter

***@***.***



9 Dec 2021, 00:46 by ***@***.***:
